crates/consensus/src/receipts/receipt.rs

```diff
@@ -519,8 +519,10 @@ pub(crate) mod serde_bincode_compat {
             // // ensure we don't have an invalid poststate variant
             data.receipt.as_receipt_mut().status = success.into();
 
-            let encoded = bincode::serialize(&data).unwrap();
-            let decoded: Data = bincode::deserialize(&encoded).unwrap();
+            let encoded = bincode::serde::encode_to_vec(&data, bincode::config::legacy()).unwrap();
```

Comment:
old one was on bincode 1.3.3, and this repo has bincode 2.01, there's breaking changes.

This code matches other parts of the code with the same pattern